core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -359,6 +360,16 @@ private void setupKnoxLoadbalancing(Stack stack, DetailedEnvironmentResponse env
         }
     }
 
+    private boolean shouldCreateOutboundLoadBalancer(boolean createPublicLb, Stack stack, LoadBalancerSku sku, DetailedEnvironmentResponse environment) {
+        return !createPublicLb
+                && AZURE.equalsIgnoreCase(stack.getCloudPlatform())
+                && LoadBalancerSku.STANDARD.equals(sku)
+                && !Optional.ofNullable(environment)
```

Comment:
Could you clarify the need for an optional here? In which cases would the environment or its internal fields be null? Just want to ensure we are not using this without reason.

Reply:
AzureEnvironmentParameters can be null and no harm be defensive here.

Reply:
I agree with being defensive but just wanted to be sure we had a reason to be, I have seen cases where we add unnecessary validation/checks when objects are guaranteed to exist and not be null.
But if the parameters can be null then this is fine.